### PR TITLE
dbft: use previous block timestamp for view 0 initialization

### PR DESCRIPTION
**Pull request by roman-khimov**
And always check new timestamp against it, timestamps can only move forward with each block.

### CODE REVIEW

File: check.go [hunk 1]
```diff
@@ -99,5 +99,5 @@ func (d *DBFT) checkChangeView(view byte) {
 		}
 	}
 
-	d.InitializeConsensus(view)
+	d.InitializeConsensus(view, 0)
```
Comment (AnnaShaleva) [inline check.go:102]: Why don't we reset it to the last accepted block timestamp?

Comment (roman-khimov) [reply check.go:102]: It stays the same in fact, but yes, we can have `d.InitializeConsensus(view, d.lastBlockTimestamp)` here as well.

File: context.go [hunk 2]
```diff
@@ -66,8 +66,9 @@ type Context struct {
 	// if this node never heard from validator i, LastSeenMessage[i] will be -1.
 	LastSeenMessage []*timer.HV
 
-	lastBlockTime  time.Time
-	lastBlockIndex uint32
+	lastBlockTimestamp uint64
+	lastBlockTime      time.Time
```
Comment (fyrchik) [inline context.go:70]: Can we add some comment here, because the difference between `Time` and `Timestamp` is not obvious?

**Review by fyrchik — APPROVED** [review]

File: check.go (hunk 1 above)
Comment (AnnaShaleva) [reply check.go:102]: Oh, I missed that `InitializeConsensus` updates timestamp only on 0 view.

Comment (fyrchik) [reply check.go:102]: I agree, `d.lastBlockTimestamp` is more straightforward and allows us to avoid `if` inside `InitializeConsensus`.

Comment (roman-khimov) [reply check.go]: Adjusted a bit (and moved this assignment to `Context`, that's where the field is anyway).

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**